### DISCUSSION TITLE
Switching to dark mode

### DISCUSSION THREAD

**Discussion by Samir-OP** [vercel/next.js#38612] [Help] [answered]
How can I switch theme to dark without using `CustomProvider`? in rsuite

**Comment by icyJoseph** [comment, marked as answer]
> If you want to change the theme of only a part of the component, just add the **.rs-theme-dark** class or the **.rs-theme-high-contrast** class to any container element to use the corresponding theme style.

I guess you can toggle this at the highest level of your app, since it is just about adding a class name. I tried it out on a codesandbox, using CRA though:

```js
const App = () => {
  const toggleClassName = () => {
    const root = document.getElementById("root"); // or __next in Next.js

    root.classList.toggle("rs-theme-dark");
  };

  return (
    <div style={styles}>
      <Button onClick={toggleClassName}>Hello World</Button>
    </div>
  );
};
```

It depends on your use case how the class name is toggled. 

Also, I didn't know about this framework, adding it to my list! Thanks!

Here's something in return, https://www.youtube.com/watch?v=UhLWAQdsgBY, perhaps, you probably should watch that video, it was very good.
